log/mod.ts

```diff
@@ -418,183 +387,10 @@ const state = {
-  msg: (T extends GenericFunction ? never : T) | (() => T),
-  ...args: unknown[]
-): T | undefined {
-  // Assist TS compiler with pass-through generic type
-  if (msg instanceof Function) {
-    return getLogger("default").warn(msg, ...args);
-  }
-  return getLogger("default").warn(msg, ...args);
-}
-
-/** Log with error level, using default logger. */
-export function error<T>(msg: () => T, ...args: unknown[]): T | undefined;
-export function error<T>(
-  msg: T extends GenericFunction ? never : T,
-  ...args: unknown[]
-): T;
-export function error<T>(
-  msg: (T extends GenericFunction ? never : T) | (() => T),
-  ...args: unknown[]
-): T | undefined {
-  // Assist TS compiler with pass-through generic type
-  if (msg instanceof Function) {
-    return getLogger("default").error(msg, ...args);
-  }
-  return getLogger("default").error(msg, ...args);
-}
-
-/** Log with critical level, using default logger. */
-export function critical<T>(msg: () => T, ...args: unknown[]): T | undefined;
-export function critical<T>(
-  msg: T extends GenericFunction ? never : T,
-  ...args: unknown[]
-): T;
-export function critical<T>(
-  msg: (T extends GenericFunction ? never : T) | (() => T),
-  ...args: unknown[]
-): T | undefined {
-  // Assist TS compiler with pass-through generic type
-  if (msg instanceof Function) {
-    return getLogger("default").critical(msg, ...args);
-  }
-  return getLogger("default").critical(msg, ...args);
-}
-
-/** Setup logger config. */
-export function setup(config: LogConfig) {
-  state.config = {
-    handlers: { ...DEFAULT_CONFIG.handlers, ...config.handlers },
-    loggers: { ...DEFAULT_CONFIG.loggers, ...config.loggers },
-  };
-
-  // tear down existing handlers
-  state.handlers.forEach((handler) => {
-    handler.destroy();
-  });
-  state.handlers.clear();
-
-  // setup handlers
-  const handlers = state.config.handlers || {};
-
-  for (const [handlerName, handler] of Object.entries(handlers)) {
-    handler.setup();
-    state.handlers.set(handlerName, handler);
-  }
-
-  // remove existing loggers
-  state.loggers.clear();
-
-  // setup loggers
-  const loggers = state.config.loggers || {};
-  for (const [loggerName, loggerConfig] of Object.entries(loggers)) {
-    const handlerNames = loggerConfig.handlers || [];
-    const handlers: BaseHandler[] = [];
-
-    handlerNames.forEach((handlerName) => {
-      const handler = state.handlers.get(handlerName);
-      if (handler) {
-        handlers.push(handler);
-      }
-    });
-
-    const levelName = loggerConfig.level || DEFAULT_LEVEL;
-    const logger = new Logger(loggerName, levelName, { handlers: handlers });
-    state.loggers.set(loggerName, logger);
-  }
-}
-
-setup(DEFAULT_CONFIG);
+export { critical } from "./critical.ts";
+export { debug } from "./debug.ts";
+export { error } from "./error.ts";
+export { getLogger } from "./get_logger.ts";
+export { info } from "./info.ts";
+export { setup } from "./_state.ts";
+export { warn } from "./warn.ts";
```

Comment:
Can you please group exports together? There's another group above. Also, please use `export * from ...` in such a way that the exports from `mod.ts` are identical to how they were before.

Reply:
Updated.